jwt_verify_lib/struct_utils.h

```diff
@@ -46,7 +46,7 @@ class StructUtils {
                            std::vector<std::string>* list);
 
  private:
-  const ::google::protobuf::Struct& struct_pb_;
+  ::google::protobuf::Struct& struct_pb_;
```

Comment:
You could use `const Struct*`  so you can set a new Struct anytime you want.  

Per my understand, that is what you need for parsing nested fields, e.g. "nested.foo.bar"


GetStruct() in line 40 is also doing a Struct copy,  which is not good.  We can change it to be
```
FindResult GetStruct(const std::string& name,
                       const ::google::protobuf::Struct* &value);
```

or
```
FindResult GetStruct(const std::string& name,
                     const  ::google::protobuf::Struct** value);
```

to pass the pointer to the Struct out.

Reply:
If we make the value `const` we won't be able to change that. Let's think about a case where I am checking for `nested.key1.key2.key3`, in this case I need to check if claim[nested], claim[nested.key1] and claim[nested.key1.key2] , the ideal way to do this is to call GetStruct(jwt_payload) in a loop and update same jwt_payload which is being passed in the GetStruct() method. This type of cases wouldn’t be possible if we make the `jwt_payload` const

Reply:
The key is NOT to make change tos `::google::protobuf::Struct` so that you don't need to make a copy of it.

Reply:
I've changed to use `const Struct*` and also add a setter `SetStructPb`.

Reply:
I prefer to add supports for Nest field names in StructUtils

* remove newly added `GetStruct`
* all `GetXYZ` functions support nested names,e.g. `a.b.c`.  

We can add following internal function to support this:

```
StructUtils::FindResult   StructUtils::findNestedField(const std::string& names, const google::protobuf::Value*& value ) {
  const auto name_vector = absl::StrSplit(names, '.');

  const auto* fields = &struct_pb_.fields();
  for (int i=0; i<name_vector.size(); ++i) {
    const auto it = fields->find(name_vector[i]);
    if (it == fields->end()) {
      return MISSING;
    }
    if (i == name_vector.size()-1) {
      value = &it->second;
      return OK;
    }
    if (it->second.kind_case() != google::protobuf::Value::kStructValue) {
      return WRONG_TYPE;
    }
    fields = &it->second.struct_value().fields();
  }
  NO_REACHABLE;
}
```


All Get functions will call this one first.

Reply:
Thanks @qiwzhang for the suggestion. This would be ideal for our use case. Let me work on this.